### PR TITLE
ign -> gz Environment Variable Migration : gz-sim

### PR DESCRIPTION
**Pull request by methylDragon**
See: https://github.com/gazebo-tooling/release-tools/issues/734

### CODE REVIEW

File: src/network/PeerTracker_TEST.cc [hunk 1]
```diff
@@ -283,11 +283,11 @@ TEST(PeerTracker, PartitionedEnv)
   gz::common::Console::SetVerbosity(4);
   EventManager eventMgr;
 
-  gz::common::setenv("IGN_PARTITION", "p1");
+  gz::common::setenv("", "p1");
```
Comment (chapulina) [inline src/network/PeerTracker_TEST.cc:286]:
```suggestion
  gz::common::setenv("GZ_PARTITION", "p1");
```

File: test/integration/log_system.cc [hunk 2]
```diff
@@ -1523,7 +1523,7 @@ TEST_F(LogSystemTest, IGN_UTILS_TEST_DISABLED_ON_WIN32(LogResources))
   // Recorded models should exist
   EXPECT_GT(entryCount(recordPath), 2);
   EXPECT_TRUE(common::exists(common::joinPaths(recordPath, homeFake,
-      ".ignition", "fuel", "fuel.ignitionrobotics.org", "openrobotics",
+      ".gz", "fuel", "fuel.ignitionrobotics.org", "openrobotics",
```
Comment (chapulina) [inline test/integration/log_system.cc:1526]: As mentioned on the `fuel-tools` PR, let's revert these for now

File: src/Server_TEST.cc [hunk 3]
```diff
@@ -849,7 +849,7 @@ TEST_P(ServerFixture, Seed)
 /////////////////////////////////////////////////
```
Comment (chapulina) [inline src/Server_TEST.cc:849]: There are more instances in this file:

```
$ grep -n "IGN_GAZEBO_RESOURCE_PATH" src/Server_TEST.cc 
1039:  common::setenv("IGN_GAZEBO_RESOURCE_PATH", "");
1074:  common::setenv("IGN_GAZEBO_RESOURCE_PATH", "");
1087:  common::setenv("IGN_GAZEBO_RESOURCE_PATH", PROJECT_SOURCE_PATH);
1099:  common::setenv("IGN_GAZEBO_RESOURCE_PATH",
```

File: include/gz/sim/Util.hh [hunk 4]
```diff
@@ -267,18 +267,18 @@ namespace gz
         Entity _entity, const EntityComponentManager &_ecm);
 
     /// \brief Environment variable holding resource paths.
-    const std::string kResourcePathEnv{"IGN_GAZEBO_RESOURCE_PATH"};
+    const std::string kResourcePathEnv{"GZ_SIM_RESOURCE_PATH"};
```
Comment (chapulina) [inline include/gz/sim/Util.hh:270]: Is this variable being tick-tocked? We should still accept `IGN_GAZEBO_RESOURCE_PATH` with warnings.

File: include/gz/sim/Util.hh [hunk 5]
```diff
@@ -267,18 +267,18 @@ namespace gz
...
     const std::string kSdfPathEnv{"SDF_PATH"};
 
     /// \brief Environment variable holding server config paths.
-    const std::string kServerConfigPathEnv{"IGN_GAZEBO_SERVER_CONFIG_PATH"};
+    const std::string kServerConfigPathEnv{"GZ_SIM_SERVER_CONFIG_PATH"};
 
     /// \brief Environment variable holding paths to custom rendering engine
     /// plugins.
-    const std::string kRenderPluginPathEnv{"IGN_GAZEBO_RENDER_ENGINE_PATH"};
+    const std::string kRenderPluginPathEnv{"GZ_SIM_RENDER_ENGINE_PATH"};
```
Comment (chapulina) [inline include/gz/sim/Util.hh:281]: I think this one isn't being ticked either

**Review by chapulina — CHANGES_REQUESTED** [review]
I found a bad comment from before, can you fix it here?

```
./src/ServerPrivate.hh:125:      /// Fuel will be checked and then the GZ_GAZEBO_RESOURCE_PATH environment
```

File: include/gz/sim/Util.hh (hunk 4 above)
Comment (methylDragon) [reply include/gz/sim/Util.hh:270]: Warning will emit via https://github.com/gazebosim/gz-common/pull/365/commits/21c47bb50f5d3397bfc1c00f647b07ac547d4fc2

https://github.com/gazebosim/gz-sim/pull/1518/commits/ac92c2ca29b222d881e5819611aec99eb26e071f

File: include/gz/sim/Util.hh (hunk 5 above)
Comment (methylDragon) [reply include/gz/sim/Util.hh:281]: Warning will emit via https://github.com/gazebosim/gz-common/pull/365/commits/21c47bb50f5d3397bfc1c00f647b07ac547d4fc2

https://github.com/gazebosim/gz-sim/pull/1518/commits/ac92c2ca29b222d881e5819611aec99eb26e071f

File: src/Server_TEST.cc (hunk 3 above)
Comment (methylDragon) [reply src/Server_TEST.cc:849]: https://github.com/gazebosim/gz-sim/pull/1518/commits/35d865fb2f5cb06269dce70704c675cea74902c7

File: src/network/PeerTracker_TEST.cc (hunk 1 above)
Comment (methylDragon) [reply src/network/PeerTracker_TEST.cc]: https://github.com/gazebosim/gz-sim/pull/1518/commits/35d865fb2f5cb06269dce70704c675cea74902c7

File: src/cmd/cmdgazebo.rb.in [hunk 6]
```diff
@@ -393,10 +393,10 @@ has properly set the DYLD_LIBRARY_PATH environment variables."
         # Check if the passed in file exists.
         if File.exists?(options['file'])
           path = options['file']
-        # If not, then first check the IGN_GAZEBO_RESOURCE_PATH environment
+        # If not, then first check the GZ_SIM_RESOURCE_PATH environment
         # variable, then the configuration path from the launch library.
         else
-          resourcePathEnv = ENV['IGN_GAZEBO_RESOURCE_PATH']
+          resourcePathEnv = ENV['GZ_SIM_RESOURCE_PATH']
```
[the remark this answers is not in the document]
Comment (methylDragon) [reply src/cmd/cmdgazebo.rb.in:399]: https://github.com/gazebosim/gz-sim/pull/1518/commits/35d865fb2f5cb06269dce70704c675cea74902c7

File: src/cmd/cmdgazebo.rb.in [hunk 7]
```diff
@@ -128,38 +128,38 @@ COMMANDS = { 'gazebo' =>
...
+  "                               GZ_SIM_RENDER_ENGINE_PATH.                   \n"\
   "\n"\
   "  --version                    Print Gazebo version information.                \n"\
   "\n"\
   "  -z [arg]                     Update rate in Hertz.                            \n"\
   "\n"+
   COMMON_OPTIONS + "\n\n" +
   "Environment variables:                                                  \n"\
-  "  IGN_GAZEBO_RESOURCE_PATH    Colon separated paths used to locate      \n"\
+  "  GZ_SIM_RESOURCE_PATH    Colon separated paths used to locate      \n"\
```
[the remark this answers is not in the document]
Comment (methylDragon) [reply src/cmd/cmdgazebo.rb.in]: https://github.com/gazebosim/gz-sim/pull/1518/commits/35d865fb2f5cb06269dce70704c675cea74902c7

File: src/SystemLoader.cc [hunk 8]
```diff
@@ -55,18 +55,38 @@ class gz::sim::SystemLoaderPrivate
     std::string homePath;
     gz::common::env(IGN_HOMEDIR, homePath);
     systemPaths.AddPluginPaths(homePath + "/.gz/sim/plugins");
+
+    // TODO(CH3): Deprecated. Remove on tock.
+    systemPaths.AddPluginPaths(homePath + "/.ignition/sim/plugins");
```
[the remark this answers is not in the document]
Comment (methylDragon) [reply src/SystemLoader.cc]: https://github.com/gazebosim/gz-sim/pull/1518/commits/35d865fb2f5cb06269dce70704c675cea74902c7

File: test/integration/log_system.cc (hunk 2 above)
Comment (methylDragon) [reply test/integration/log_system.cc]: https://github.com/gazebosim/gz-sim/pull/1518/commits/3417cef701d64d734dadaf3d2bafd1754e89b383

File: include/gz/sim/Util.hh (hunk 4 above)
Comment (chapulina) [reply include/gz/sim/Util.hh:270]: Smart!

File: src/rendering/RenderUtil.cc [hunk 9]
```diff
@@ -2504,7 +2504,27 @@ void RenderUtil::Init()
     return;
 
   gz::common::SystemPaths pluginPath;
-  pluginPath.SetPluginPathEnv(kRenderPluginPathEnv);
+
+  // TODO(CH3): Deprecated. Remove on tock.
+  std::string result;
+  if (!env(kRenderPluginPathEnv, result))
```
Comment (chapulina) [inline src/rendering/RenderUtil.cc:2510]: Compilation is failing:

```
[01m[K/home/jenkins/workspace/ignition_gazebo-ci-pr_any-ubuntu_auto-amd64/ign-gazebo/src/rendering/RenderUtil.cc:2510:8:[m[K [01;31m[Kerror: [m[K‘[01m[Kenv[m[K’ was not declared in this scope; did you mean ‘[01m[Kgz::common::env[m[K’?
```

Comment (methylDragon) [reply src/rendering/RenderUtil.cc:2510]: :thinking:

Comment (methylDragon) [reply src/rendering/RenderUtil.cc]: https://github.com/gazebosim/gz-sim/pull/1518/commits/3c17b825855880269b660e3e34dcdf1e8062c0e8

**Comment by chapulina** [comment]
Some of the test failures are new, compare to the failures on `main`:

https://build.osrfoundation.org/job/ignition_gazebo-ci-main-focal-amd64/33/#showFailuresLink

**Comment by methylDragon** [comment]
> Some of the test failures are new, compare to the failures on `main`:
> 
> [build.osrfoundation.org/job/ignition_gazebo-ci-main-focal-amd64/33/#showFailuresLink](https://build.osrfoundation.org/job/ignition_gazebo-ci-main-focal-amd64/33/#showFailuresLink)

I ran this again locally, and that test passed :cold_sweat: 
I'm just waiting for CI to rerun the test..
image

**Comment by methylDragon** [comment]
A new flaky test perhaps?

https://build.osrfoundation.org/job/ignition_gazebo-ci-pr_any-ubuntu_auto-amd64/8855/testReport/

**Review by chapulina — APPROVED** [review]
> A new flaky test perhaps?

huh could be, I think it may have needed one of the upstream libs' nightlies 🤔 Well, it's all good now!

**Pull request merged**